crates/core/src/factories/mem_op_store.rs

```diff
@@ -189,9 +189,17 @@ impl OpStore for Kitsune2MemoryOpStore {
                 K2Error::other_src("Failed to deserialize op data, are you using `Kitsune2MemoryOp`s?", e)
             })?;
 
-            let op_ids =
-                ops_to_add.iter().map(|(op_id, _)| op_id.clone()).collect();
-            self.write().await.op_list.extend(ops_to_add);
+            let mut op_ids = Vec::with_capacity(ops_to_add.len());
```

Comment:
We shouldn't overwrite when inserting into the op store. I could swear I already fixed this but clearly not.